src/tests/ConfigBuilder.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Signal Estimator authors
+// Licensed under MIT
+
+#pragma once
+
+#include "core/Config.hpp"
+
+class ConfigBuilder {
```

Comment:
I think introducing ConfigBuilder/DevInfoBuilder is an overkill here. The cost is the whole new abstraction layer between tests and tested code, however I see no big benefits to be worth it.

Can't we just use aggregate initialization here?

Like:

```cpp
Config { .foo = 123, .bar = 456 }
```

Reply:
I agree, the builder pattern has quite some overhead and I would love to get rid of it. I would prefer your suggested solution as well, but it requires C++20 and this project currently is still on C++17

Reply:
I think C++20 is a bit too early for us, since we also target embedded computers which are often more outdated.

However in nearest future we anyway support only gcc and clang (because we don't support anything except ALSA/linux yet), and they seem to support it fine:

```
$ cat 1.cpp
#include <iostream>

struct S {
    int x = 123;
    int y = 0;
};

int main() {
    S s = S { .y = 456 };
    std::cout << s.x << s.y << "\n";
    return 0;
}

$ g++ -std=c++17 1.cpp && ./a.out 
123456

$ clang++ -std=c++17 1.cpp && ./a.out
123456
```

Reply:
I will check if it still compiles with `set(CMAKE_CXX_EXTENSIONS OFF)` and then get rid of the builders if it still compiles using CMake.

Reply:
IIRC gnu extensions are anyway required by libasound headers, so we can use them. By the time when we will support anything except gcc/clang and linux/macos (if it ever happens), we probably will just switch to c++20 :)

Reply:
Done. Replaced the builders with aggregate initialization using designated initializers.